Sources/DebugExtensions/DebugProxy+Export.swift

```diff
@@ -0,0 +1,225 @@
+
+    /// Opens a system dialog for picking a folder where to export the rendering artifacts
+    /// (the image as TIFF and various rendering graphs as PDFs) to.
+    /// - Parameter filePrefix: A prefix that is added to the files. By default, the name of the app is used.
+    func export(filePrefix: String? = nil) {
+        let filePrefix = exportFilePrefix(filePrefix: filePrefix)
+        openSavePanel(message: "Select folder where to save the render results") { url in
+            // Write rendering results into the picked folder.
+            self.writeResults(to: url, with: filePrefix)
+        }
+    }
+
+}
+
+/// Opens the system panel for selecting a folder to save rendering results to.
+/// - Parameters:
+///   - message: A message to display on top of the panel.
+///   - callback: A callback for writing the files to the chosen URL.
+private func openSavePanel(message: String, callback: @escaping (URL) -> Void) {
+    // Use the system panel for picking the folder where to save the files.
+    let openPanel = NSOpenPanel()
+    openPanel.message = message
+    openPanel.prompt = "Select"
+    openPanel.canChooseFiles = false
+    openPanel.canChooseDirectories = true
+    openPanel.canCreateDirectories = true
+
+    openPanel.begin { response in
+        guard response == .OK else { return }
+        callback(openPanel.url!)
+    }
+}
+
+#endif // AppKit
+
+
+// MARK: - iOS
+
+#if canImport(UIKit)
+
+import UIKit
+
+public extension CIImage.DebugProxy {
+
+    /// Renders the image with the given settings and opens a share sheet for exporting the image.
+    /// - Parameters:
+    ///   - filePrefix: A prefix that is added to the image file. By default, the name of the app is used.
+    ///   - codec: The codec of the exported image. Uses uncompressed TIFF by default.
+    ///   - format: The image format. This influences bit depth (8-, 16-, or 32-bit) and pixel format (uint or float). Defaults to 8-bit uint.
+    ///             Note that not all formats are supported by all codecs. E.g., `jpeg` and `heif` only supports 8-bit formats.
+    ///   - quality: The quality of the exported image, i.e., the amount of compression. Only supported by `jpeg`, `heif`, and `heif10` codecs.
+    ///   - colorSpace: The color space of the exported image. By default, the Display P3 color space is used.
+    ///                 Note that it needs to match the chosen `format`, i.e., a single-channel format needs a grayscale color space.
+    func export(filePrefix: String? = nil, codec: ImageCodec = .tiff, format: CIFormat = .RGBAh, quality: Float = 1.0, colorSpace: CGColorSpace? = nil) {
+        let filePrefix = exportFilePrefix(filePrefix: filePrefix)
+        let fileURL = FileManager.default.temporaryDirectory.appendingPathComponent("\(filePrefix)_image").appendingPathExtension(codec.fileExtension)
+        self.write(to: fileURL, codec: codec, format: format, quality: quality, colorSpace: colorSpace)
+        openShareSheet(for: [fileURL])
```

Comment:
`openShareSheet` silently fell through on my Mac M1 (Ventura 13.2.1), i.e., no execution and message. Not sure if this is suppose to happen, the files previously exported were all there in a temporary app folder.

Reply:
To clarify: This ☝️ was happening when running tests, which doesn't seem to support showing file pickers. It works normally when run inside an app.